common/src/main/java/org/ergoplatform/appkit/Address.java

```diff
@@ -86,7 +85,7 @@ public NetworkType getNetworkType() {
      * @throws IllegalArgumentException if this instance is not P2PK address
      */
     public P2PKAddress asP2PK() {
-        checkArgument(isP2PK(), "This instance %s is not P2PKAddress", this);
+        InternalUtil.checkArgument(isP2PK(), String.format("This instance %s is not P2PKAddress", this));
```

Comment:
The previous code avoided String.format to be called when no errors.

Reply:
Is it it ok to just use a string concatenation then